qrcode/main.go

```diff
@@ -41,12 +43,25 @@ Usage:
 	if *size <= 0 {
 		checkError(fmt.Errorf("Error: value of -s should > 0"))
 	}
-	if len(flag.Args()) == 0 {
-		flag.Usage()
-		checkError(fmt.Errorf("Error: no content given"))
-	}
 
 	content := strings.Join(flag.Args(), " ")
+	if *inFile != "" {
```

Comment:
I suggest prioritising the -i flag, then command line arguments, then reading from STDIN last, like:

	content := ""
	argsContent := strings.Join(flag.Args(), " ")

	if *inFile != "" {
		b, err := ioutil.ReadFile(*inFile)
		if err != nil {
			checkError(err)
		}
		content = string(b)
	} else if argsContent != "" {
		content = argsContent
	} else {
		b, err := ioutil.ReadAll(os.Stdin)
		if err != nil {
			checkError(err)
		}
		content = string(b)
	}